great_tables/_utils_render_html.py

```diff
@@ -719,9 +719,17 @@ def _get_table_defs(data: GTData) -> dict[str, Any]:
 
     # Get all the widths for the columns as a list where None values mean that the width is
     # not set for that column
-    # TODO: ensure that the stub column is set first in the list
     widths = data._boxhead._get_column_widths()
 
+    # Get the stub column variable
+    stub_var = data._boxhead._get_stub_column()
+
+    # If there is a stub column, then get the width for that column and insert it at the beginning
+    if stub_var is not None:
+
+        stub_width = data._boxhead._get_column_width_by_var(var=stub_var.var)
```

Comment:
Could we add a method like `Boxhead.canonical_columns()` or `.final_columns()`, that returns a list[ColInfo] for all final visible columns (including stub, and group column if correct option is set), in the order they will be rendered in?

This way, Boxhead becomes the source of truth on the final columns. I realize there's some ambiguity between Stub and Boxhead columns. But it seems okay for the final columns across these two things to live in one place, and the Boxhead seems like it might be the best fit?

In this case, the code could just call something like...

```
columns = data._boxhead.canonical_columns(options)

# widths for all columns that will be represented as columns
# e.g. groupings are rows by default, so wouldn't be included,
# but if the option to make them a column is set, then they would
widths = [col.width for col in columns]
```